server/api/assets.js

```diff
@@ -81,42 +86,26 @@ router.post(
       if (!isTagsValid(tags)) throw new ValidationError('tags length invalid');
 
       // check license
-      if (!license) throw new ValidationError('asset need a license');
-      if (!isLicenseValid(license))
+      if (!license || !licenseId) {
+        throw new ValidationError('asset need a license');
+      }
+      if (!isLicenseValid(licenseId, license))
         throw new ValidationError('license invalid');
 
       // check asset
       const { file: asset } = req;
       const hash256 = sha256(asset.buffer);
       const { height, width } = imageSize(asset.buffer);
-      validateImage(asset, assetSHA256);
```

Comment:
Should check image hash equal

Reply:
`likeFingerprint` is the hash from client. It is validated in the function that server calculates again.

Reply:
As `hash256` already calculated 2 lines above, you could pass it into `validateImage()`.